src/parser/GenericParser.test.ts

```diff
@@ -0,0 +1,281 @@
+import { expect, jest } from "@jest/globals";
+import { GenericParser } from "./GenericParser";
+import { findFilesAsync } from "./helper/Helper";
+import { TreeParser } from "./helper/TreeParser";
+import { getParserConfiguration } from "../../test/metric-end-results/TestHelper";
+import { Language, languageToGrammar } from "./helper/Language";
+import Parser from "tree-sitter";
+import { FileMetric } from "./metrics/Metric";
+import { MetricCalculator } from "./MetricCalculator";
+
+jest.mock("./CouplingCalculator");
+jest.mock("./helper/Helper");
+jest.mock("./helper/TreeParser");
+jest.mock("./MetricCalculator");
+
+async function* mockedFindFilesAsync() {
+    yield { fileExtension: "cpp", filePath: "clearly/invalid/path.cpp" };
+}
+
+async function* mockedFindTwoFiles() {
+    yield { fileExtension: "cc", filePath: "clearly/invalid/path1.cc" };
+    yield { fileExtension: "cpp", filePath: "clearly/invalid/path2.cpp" };
+}
+
+async function* mockedErrorFindFilesAsync() {
+    yield { fileExtension: "cpp", filePath: "clearly/invalid/path.cpp" };
+    throw new Error("Hard drive crashed!");
+}
+
+let parser: Parser;
+let tree: Parser.Tree;
+
+beforeAll(() => {
+    parser = new Parser();
+    parser.setLanguage(languageToGrammar.get(Language.CPlusPlus));
+    tree = parser.parse("int main() { return 0; }");
+});
+
+beforeEach(() => {
+    // Clear all instances and calls to constructor and all methods:
+    jest.clearAllMocks();
+    jest.resetAllMocks();
+});
+
+describe("GenericParser.calculateMetrics() unit tests", () => {
```

Comment:
The combination of describe and it blocks can help you to have a human readable message in the test outputs. In this case it would be `GenericParser.calculateMetrics() unit tests should return the correct value and call the appropriate functions in case there is no error`. Maybe it would be better to have something like describe: `GenericParser`, it: `should calculate lines of code metric`

Reply:
Changed the naming